packages/workshop-app/app/utils/launch-editor.server.ts

```diff
@@ -291,44 +301,79 @@ export async function launchEditor(
-		// handle the path translation, but only if a relative path is used.
-		fileName = path.relative('', fileName)
+	if (typeof pathList === 'string') {
+		pathList = [pathList]
 	}
 
-	// cmd.exe on Windows is vulnerable to RCE attacks given a file name of the
-	// form "C:\Users\myusername\Downloads\& curl 172.21.93.52". Use a whitelist
-	// to validate user-provided file names. This doesn't cover the entire range
-	// of valid file names but should cover almost all of them in practice.
-	if (
-		process.platform === 'win32' &&
-		!WINDOWS_FILE_NAME_WHITELIST.test(
-			// replacing characters we know are fine (because heck if I'm going to edit that regex above 🙃)
-			fileName.replace(/\+|\$|\[|\]/g, '').trim(),
-		)
-	) {
-		return {
-			status: 'error',
-			error: `Could not open ${fileName} in the editor. When running on Windows, file names are checked against a whitelist to protect against remote code execution attacks. File names may consist only of alphanumeric characters (all languages), periods, dashes, slashes, and underscores.`,
-		}
+	type accumulator = {
+		fileList: string[]
+		errorsList: string[]
 	}
 
-	if (!fs.existsSync(fileName)) {
-		fsExtra.ensureDirSync(path.dirname(fileName))
-		fsExtra.writeFileSync(fileName, '', 'utf8')
+	const initArgs: accumulator = { fileList: [], errorsList: [] }
+
+	const { fileList, errorsList } = pathList.reduce(
+		(acc: accumulator, fileName: string) => {
+			if (
+				process.platform === 'linux' &&
+				fileName.startsWith('/mnt/') &&
+				/Microsoft/i.test(os.release())
+			) {
+				// Assume WSL / "Bash on Ubuntu on Windows" is being used, and
+				// that the file exists on the Windows file system.
+				// `os.release()` is "4.4.0-43-Microsoft" in the current release
+				// build of WSL, see: https://github.com/Microsoft/BashOnWindows/issues/423#issuecomment-221627364
+				// When a Windows editor is specified, interop functionality can
+				// handle the path translation, but only if a relative path is used.
+				fileName = path.relative('', fileName)
+			}
+
+			// cmd.exe on Windows is vulnerable to RCE attacks given a file name of the
+			// form "C:\Users\myusername\Downloads\& curl 172.21.93.52". Use a whitelist
+			// to validate user-provided file names. This doesn't cover the entire range
+			// of valid file names but should cover almost all of them in practice.
+			if (
+				process.platform === 'win32' &&
+				!WINDOWS_FILE_NAME_WHITELIST.test(
+					// replacing characters we know are fine (because heck if I'm going to edit that regex above 🙃)
+					fileName.replace(/\+|\$|\[|\]/g, '').trim(),
+				)
+			) {
+				acc.errorsList.push(fileName)
+			} else {
+				if (!fs.existsSync(fileName)) {
+					fsExtra.ensureDirSync(path.dirname(fileName))
+					fsExtra.writeFileSync(fileName, '', 'utf8')
+				}
+
+				acc.fileList.push(fileName.trim())
+			}
+			return acc
+		},
+		initArgs,
+	)
+
+	function getErrorMessage() {
+		let error: string
+		if (errorsList.length) {
```

Comment:
What is it about errorsList.length that tells us the error is going to be related the windows allowlist functionality? Couldn't this also be a result of something else?

Reply:
NO 
the code used to return error only in two cases:
 - if the path was not valid in windows
 - _childProcess to launch the editor exit with non zero value

Reply:
But what if in the future we make a change to this file and there's another way to have errors? I really don't like the indirection here.

Reply:
I'm still concerned about this.

Reply:
the only error that we check for are for invalid windows files.
I case you clicked `Open All Files` we will open all valid files and show an error message if there are any invalid file